pkg/alertmanager/amcfg.go

```diff
@@ -376,12 +412,15 @@ func (cg *configGenerator) convertWebhookConfig(ctx context.Context, in monitori
 	}
 
 	if in.URLSecret != nil {
-		url, err := cg.store.GetSecretKey(ctx, crKey.Namespace, *in.URLSecret)
+		url, err := cg.getValidURLFromSecret(ctx, crKey.Namespace, *in.URLSecret)
 		if err != nil {
-			return nil, errors.Errorf("failed to get key %q from secret %q", in.URLSecret.Key, in.URLSecret.Name)
+			return nil, errors.Wrap(err, "convertWebhookConfig failed")
```

Comment:
I think that the caller already wraps the error indicating that it's a WebhookConfig resource.